gpt_engineer/core/default/steps.py

```diff
@@ -0,0 +1,257 @@
+    set up and functional. Ensure these prerequisites are in place before invoking `simple_gen`.
+    """
+    db = FileRepository(PREPROMPTS_PATH)
+    messages = ai.start(setup_sys_prompt(db), prompt, step_name=curr_fn())
+    chat = messages[-1].content.strip()
+    memory[CODE_GEN_LOG_FILE] = chat
+    files = parse_chat(chat)
+    workspace = FileRepository(workspace_path)
+    for file_name, file_content in files:
+        workspace[file_name] = file_content
+    code = Code({key: val for key, val in files})
+    return code
+
+
+def gen_entrypoint(workspace_path: str, ai: AI, memory: FileRepository) -> Code:
+    """
+    Generates an entry point script based on a given codebase's information.
+
+    This function prompts the AI model to generate a series of Unix terminal commands
+    required to a) install dependencies and b) run all necessary components of a codebase
+    provided in the workspace. The generated commands are then saved to 'run.sh' in the
+    workspace.
+
+    Parameters:
+    - ai (AI): An instance of the AI model.
+    - dbs (DBs): An instance containing the database configurations and workspace
+      information, particularly the 'all_output.txt' which contains details about the
+      codebase on disk.
+
+    Returns:
+    - List[dict]: A list of messages containing the AI's response.
+
+    Notes:
+    - The AI is instructed not to install packages globally, use 'sudo', provide
+      explanatory comments, or use placeholders. Instead, it should use example values
+      where necessary.
+    - The function uses regular expressions to extract command blocks from the AI's
+      response to create the 'run.sh' script.
+    - It assumes the presence of an 'all_output.txt' file in the specified workspace
+      that contains information about the codebase.
+    """
+    messages = ai.start(
+        system=(
+            "You will get information about a codebase that is currently on disk in "
+            "the current folder.\n"
+            "From this you will answer with code blocks that includes all the necessary "
+            "unix terminal commands to "
+            "a) install dependencies "
+            "b) run all necessary parts of the codebase (in parallel if necessary).\n"
+            "Do not install globally. Do not use sudo.\n"
+            "Do not explain the code, just give the commands.\n"
+            "Do not use placeholders, use example values (like . for a folder argument) "
+            "if necessary.\n"
+        ),
+        user="Information about the codebase:\n\n" + memory[CODE_GEN_LOG_FILE],
+        step_name=curr_fn(),
+    )
+    print()
+
+    regex = r"```\S*\n(.+?)```"
+    matches = re.finditer(regex, messages[-1].content.strip(), re.DOTALL)
+    entrypoint_code = Code({ENTRYPOINT_FILE: "\n".join(match.group(1) for match in matches)})
+    # write entrypoint code to file
+    for key, val in entrypoint_code.items():
+        FileRepository(workspace_path)[key] = val
```

Comment:
Same here, I don't see why we write to disk here

Reply:
See my comment below - but if we make a rule that core library can not reference the data library then you will guarantee nothing will get written to disk by the core library